### PR TITLE
add a color shifting variant of the floor light

### PR DESCRIPTION
**Pull request by TheLordME**
## Why It's Good For The Game
more than one color!

## Changelog

:cl:
add: Added new floor light version that changes colors every 0.5 seconds
/:cl:

### CODE REVIEW

**Comment by TheLordME** [comment]
https://user-images.githubusercontent.com/58342752/229480531-62393fee-1816-497b-a52c-8bbd5b800222.mov

File: code/game/machinery/floor_light.dm [hunk 1]
```diff
@@ -171,3 +173,44 @@ var/list/floor_light_cache = list()
 /obj/machinery/floor_light/cultify()
 	default_light_colour = "#FF0000"
 	update_brightness()
+
+/obj/machinery/floor_light/changing
+	name = "changing Floor light"
```
Comment (Lohikar) [inline code/game/machinery/floor_light.dm:178]: nit: `changing floor light`

File: code/game/machinery/floor_light.dm [hunk 2]
```diff
@@ -171,3 +173,44 @@ var/list/floor_light_cache = list()
...
+
+/obj/machinery/floor_light/changing/Initialize(mapload)
+	. = ..()
+	last_color_change = world.time
+
+/obj/machinery/floor_light/changing/process(delta_time)
+	. = ..()
+	update_colour()
+
+/obj/machinery/floor_light/changing/proc/update_colour()
```
Comment (Lohikar) [inline code/game/machinery/floor_light.dm:189]: nit: should probably be `update_color` for consistency

File: code/game/machinery/floor_light.dm [hunk 3]
```diff
@@ -171,3 +173,44 @@ var/list/floor_light_cache = list()
...
+	if(last_color_change > world.time - 0.5 SECONDS)
```
Comment (Lohikar) [inline code/game/machinery/floor_light.dm:190]: Process only runs every two seconds on machinery, no?

Comment (TheLordME) [reply code/game/machinery/floor_light.dm:190]: Dunno, but either way enough
It doesn't exactly matter how fast it changes, just often enough to be noticable

File: code/game/machinery/floor_light.dm (hunk 2 above)
Comment (TheLordME) [reply code/game/machinery/floor_light.dm:189]: Fair

File: code/game/machinery/floor_light.dm (hunk 1 above)
Comment (TheLordME) [reply code/game/machinery/floor_light.dm:178]: Fair

**Review by Captain277 — APPROVED** [review]

**Pull request merged**